### DISCUSSION TITLE
Target object has not been processed because no matching rules were found, but starts analyzing right after?

### DISCUSSION THREAD

**Discussion by BenjaminEngeset** [Azure/PSRule.Rules.Azure#1422] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/99641908/168464469-e93a71e1-7092-424f-9f7a-73ee0d1b14ac.png)

Then 

![image](https://user-images.githubusercontent.com/99641908/168464494-548ca427-078c-4434-881f-55ed5bad4ca1.png)

What is causing this? Build always starts like this for all modules, but still analyzes them right after. For information, main.test.bicep is just referring to main.bicep, but with required parameters.

![image](https://user-images.githubusercontent.com/99641908/168464589-cd24645b-97e2-4a47-a12b-12a74736bdca.png)

**Comment by BernieWhite** [comment, marked as answer]
@bengeset96 PSRule for Azure doesn't include any rules that test the Bicep files directly. Instead we export the Azure resources, which is the reason for the warning. We could do better explaining this and providing a different default behaviour.

An examples of a rule that does target the file would be:

- Looking for a copyright header. [example](https://github.com/microsoft/PSRule.Rules.MSFT.OSS/blob/9a828123f5e9218edd947b6ef4efd2deb53b36d8/src/PSRule.Rules.MSFT.OSS/MSFT.OSS.Rule.ps1)
- Checking naming convention of the Bicep file to be `main.bicep` or looking for a readme in the same path.

---

Your code is fine. If you want to disable the warning configure the configure the `Execution.NotProcessedWarning` option.

For example, you can do this in `ps-rule.yaml` by:

```yaml
execution:
  notProcessedWarning: false
```

Other options on how to configure this are available in the [docs](https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Options/#executionnotprocessedwarning).

I hope that helps.

**Reply by BenjaminEngeset** [reply]
It makes sense!

Thank you for the clarification.

I added this yesterday and it solved all my cosmetic "issues".

```
execution:
  notProcessedWarning: false
```
